server/server.go

```diff
@@ -685,6 +685,8 @@ func (s *Server) getTLSConfig() *tls.Config {
 func killConn(conn *clientConn) {
 	sessVars := conn.ctx.GetSessionVars()
 	atomic.StoreUint32(&sessVars.Killed, 1)
+	// 'Kill' status can be showed in Command/Info field when show processlist
+	conn.ctx.SetProcessInfo("this session is being killed", time.Now(), mysql.ComProcessKill, 0)
```

Comment:
Ideally, for MySQL compatibility the `Info` column is not overwritten. This helps you see what is being actively killed. The State should also say "Killed" instead of autocommit.

For #24031, there are really two issues:
1. Update the status for connections which have been "killed" so that it is more accurate.
2. Change the `wait_timeout` to be a loop, so the connection which is being killed can cleanup faster.

This fixes issue (1), but we should also fix (2) to consider #24031 resolved.

Reply:
thx for your review, and do we need design doc for fix (2)?